src/www/firewall_nat_edit.php

```diff
@@ -256,16 +255,33 @@
             $natent['natreflection'] = $pconfig['natreflection'];
         }
 
-        // If we used to have an associated filter rule, but no-longer should have one
-        if (isset($id) && !empty($a_nat[$id]['associated-rule-id']) && ( empty($natent['associated-rule-id']) || $natent['associated-rule-id'] != $a_nat[$id]['associated-rule-id'] ) ) {
-            // Delete the previous rule
-            foreach ($config['filter']['rule'] as $key => $item){
-                if(isset($item['associated-rule-id']) && $item['associated-rule-id']==$a_nat[$id]['associated-rule-id'] ){
-                    unset($config['filter']['rule'][$key]);
-                    break;
+        if (isset($id) && !empty($a_nat[$id]['associated-rule-id'])) {
+            // If we used to have an associated filter rule, but no-longer should have one
+            if ((empty($natent['associated-rule-id']) || $natent['associated-rule-id'] != $a_nat[$id]['associated-rule-id'])) {
+                // Delete the previous rule
+                foreach ($config['filter']['rule'] as $key => $item) {
+                    if (isset($item['associated-rule-id']) && $item['associated-rule-id'] == $a_nat[$id]['associated-rule-id']) {
+                        unset($config['filter']['rule'][$key]);
+                        break;
+                    }
                 }
+                mark_subsystem_dirty('filter');
+            }
+
+            // toggle nat rule and associated rule if it exists
+            if (isset($a_nat[$id]["disabled"]) != $natent['disabled']) {
```

Comment:
not sure this is a good idea, it could lead to side affects. When a user disables an associated filter rule, it will automatically re-enable on save here. For an explicit toggle it might work, but I'm really in doubt if we should do this (people clone and disable associated rules). @fichtner what do you think?